src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const keyValue = item[key];
+
+    if (!((keyValue as unknown as string) in result)) {
+      result[keyValue as unknown as string] = [];
+    }
+  }
```

Comment:
The first loop is used to initialize the `result` object with empty arrays. This can be optimized by combining the initialization with the pushing of items into the arrays in the second loop, thus eliminating the need for two separate loops over `items`.